docs/DeveloperGuide.md

```diff
@@ -346,14 +415,229 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
-    * 3a1. AddressBook shows an error message.
+    * 3a1. ContactCS requests the user to provide the correct index
+    * 3a2. User enters new index
+    * Steps 3a1 - 3a2 are repeated until the index entered are correct
+
+      Use case resumes from step 4.
+
+* 4a. The given format is invalid.
+
+    * 4a1. ContactCS requests the user to provide the correct format and shows the valid command format
+    * 4a2. User enters new input
+    * Steps 4a1 - 4a2 are repeated until the format for the entered input is correct
+
+      Use case resumes from step 4.
+
+**Use case: Mark frequently used contacts**
+
+**MSS**
+
+1.  User requests to list/search persons
+2.  ContactCS shows a list of persons
+3.  User requests to mark certain contacts as frequently accessed
+4.  ContactCS marks these contacts for quick access
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. The list is empty.
+  * 2a1. ContactCS displays a message to the user saying that the list is empty
+
+      Use case ends.
+
+* 3a. The given index is invalid.
+
+    * 3a1. ContactCS requests the user to provide the correct index
+    * 3a2. User enters new index
+    * Steps 3a1 - 3a2 are repeated until the index entered are correct
+
+      Use case resumes from step 4.
+
+* 4a. The given format is invalid.
+
+    * 4a1. ContactCS requests the user to provide the correct format and shows the valid command format
+    * 4a2. User enters new input
+    * Steps 4a1 - 4a2 are repeated until the format for the entered input is correct
+
+      Use case resumes from step 4.
+
+**Use case: Access frequently used contacts**
+
+**MSS**
+
+1.  User requests to access frequently used contacts
+2.  ContactCS shows a list of persons
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. The list is empty.
+  * 2a1. ContactCS displays a message to the user saying that the list is empty
+
+    Use case ends.
+
+* 3a. The given format is invalid.
+
+    * 3a1. ContactCS requests the user to provide the correct format and shows the valid command format
+    * 3a2. User enters new input
+    * Steps 3a1 - 3a2 are repeated until the format for the entered input is correct
+
+      Use case resumes from step 2.
+
+**Use case: View usage instructions**
+
+**MSS**
+
+1.  User requests to view usage instructions for the app
+2.  ContactCS displays the usage instructions and commands available for the user
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. The given format is invalid.
+
+    * 2a1. ContactCS requests the user to provide the correct format and shows the valid command format
+    * 2a2. User enters new input
+    * Steps 2a1 - 2a2 are repeated until the format for the entered input is correct
+
+      Use case resumes from step 2.
+
+**Use case: View app with sample data**
+
+**MSS**
+
+1.  User requests to view app with sample data
+2.  ContactCS populates the interface with sample data for demonstration purposes
+3.  User interacts with the app to see how it functions with the sample data
```

Comment:
Not sure if step 3 is necessary, but I'll just leave this for the rest to decide

Reply:
hmmm now you point it out I think it might be redundant, cause even I don't mention this line the person read it still can get to it implicitly from step 2. So I will remove it.